pkgs/by-name/pr/prometheus-tailscale-exporter/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  nixosTests,
+  versionCheckHook,
+}:
+
+buildGoModule rec {
+  pname = "tailscale-exporter";
+  version = "0.2.5";
+
+  src = fetchFromGitHub {
+    owner = "adinhodovic";
+    repo = "tailscale-exporter";
+    tag = version;
+    hash = "sha256-6iQtGfQsXVmwFaSA7B1AG+kbtSyKVWFbEld1lMb0DnE=";
+  };
+
+  patches = [
+    # This patch prevents the exporter from exiting immediately upon
+    # startup when no credentials are provided, which is useful for
+    # testing the NixOS module.
+    ./allow-running-without-credentials.patch
```

Comment:
We should probably only apply this inside the NixOS test.

Reply:
Fixed. This is indeed a good improvement :+1: I had initially wanted to do this myself but was lazy to figure out the pieces to do this. In the end I think I found a good way. Let me know what you think.